pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -147,6 +152,25 @@ func (pc *PodCache) getPodByIP(addr string) *v1.Pod {
 	return item.(*v1.Pod)
 }
 
+// getPod first checks if a pod with ip exists in the cache. If it exists, it just returns
+// from the cache otheriwse it loads the pod from underlying store.
+func (pc *PodCache) getPod(addr string, namespace string) *v1.Pod {
+	pod := pc.getPodByIP(addr)
+	if pod != nil {
+		return pod
+	}
+	pods, err := pc.lister.Pods(namespace).List(k8slabels.NewSelector())
```

Comment:
This may not make sense, this get pod from the local cache.

Reply:
Sorry, I did not fully read your comment on the issue, I think you were mentioning the same there any list on informer, would get it from local cache?

Reply:
Changed, Does the new call get directly from K8s?